shared/src/main/scala/squants/Quantity.scala

```diff
@@ -308,8 +308,3 @@ abstract class Quantity[A <: Quantity[A]] extends Serializable with Ordered[A] {
   def map(f: Double ⇒ Double): A = unit(f(value))
 }
 
-abstract class StrictlyPositiveQuantity[A <: StrictlyPositiveQuantity[A]](value: Double) extends Quantity[A] {self: A ⇒
```

Comment:
The problem is that we run into complications such as these: Take the scenario `KilogramsMetersSquared(1) - KilogramsMetersSquared(2) + KilogramsMetersSquared(2)`
After the first subtraction, we get a negative moment, resulting in `None`. The addition afterwards would add a `None` with a `KilogramsMetersSquared(2)`, getting `None`
However, mathematically, this should be evaluated to the equivalent of `Some(KilogramsMetersSquared(1))`.  Implementing this kind of feature is certainly possible, but scenarios such as these make it complicated and outside the scope of this pr.